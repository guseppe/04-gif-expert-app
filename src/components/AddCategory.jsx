import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories, onAddCategory}) => {
  const [inputValue, setInputValue ] = useState('');
  const onInputChange = ({target}) =>{
    setInputValue(target.value);
  }
  const onSubmit = (event) =>{
    event.preventDefault();
    //console.log(inputValue);
    let value = inputValue.trim();
    if(value.length <= 1) return;
    //setCategories(categories => [...categories, inputValue]); esta una forma de comunicar data al padre
    onAddCategory(value);
    setInputValue('');
  }
  return (
      <form onSubmit={ onSubmit } aria-label="categoryForm">
        <input 
          type="text"
          placeholder="Buscar Gifts"
          value={ inputValue }
          onChange={ onInputChange }
        />
      </form>
  )
}

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired,
}