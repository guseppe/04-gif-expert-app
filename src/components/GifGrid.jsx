import { useEffect, useState } from "react";
import { getGifs } from "../helpers/GetGifs"

export const GifGrid = ({category}) => {
  const [counter,setCounter] = useState(0);  
  useEffect(()=>{
    getGifs(category);
  }, [] );
  //getGifs(category);
  return (
    <>
      <h3>{category}</h3>
      {/* <h5>{counter}</h5>
      <button onClick={()=>setCounter(counter + 1)}>+1</button> */}
    </>
  )
}
