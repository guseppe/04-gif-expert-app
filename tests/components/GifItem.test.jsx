import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('Test in <GifItem/>', () => {
  
  const title = 'Dragon Ball GIF by Toei Animation';
  const url = 'https://media0.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=6f627e39y19ungcan8ud6bcsg1ummryhpr2xvibmjsft10qo&ep=v1_gifs_search&rid=giphy.gif&ct=g';
  test('Should render a gif item', () => {
    const { container } = render(<GifItem title ={title} url ={url}/>);
    expect(container).toMatchSnapshot();
  })
})

