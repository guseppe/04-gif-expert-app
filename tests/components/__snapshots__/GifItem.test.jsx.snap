// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> Should render a gif item 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball GIF by Toei Animation"
      src="https://media0.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=6f627e39y19ungcan8ud6bcsg1ummryhpr2xvibmjsft10qo&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon Ball GIF by Toei Animation
    </p>
  </div>
</div>
`;
